src/apps/content-editor/src/app/views/ItemEdit/Meta/index.tsx

```diff
@@ -272,16 +422,51 @@ export const Meta = forwardRef(
                   media content in the preview on the right.
                 </Typography>
               </Box>
-              <MetaTitle
-                value={web.metaTitle}
-                onChange={handleOnChange}
-                error={errors?.metaTitle}
-              />
-              <MetaDescription
-                value={web.metaDescription}
-                onChange={handleOnChange}
-                error={errors?.metaDescription}
-              />
+              <AIGeneratorParameterProvider>
+                <MetaTitle
+                  value={web.metaTitle}
+                  onChange={handleOnChange}
+                  error={errors?.metaTitle}
+                  saveMetaTitleParameters={flowType === FlowType.AIGenerated}
+                  onResetFlowType={() => {
+                    if (flowType === FlowType.AIGenerated) {
+                      setFlowType(FlowType.Manual);
+                    }
+                  }}
+                  onAIMetaTitleInserted={() => {
+                    // Scroll to and open the meta description ai generator to continue
+                    // with the AI-assisted flow
+                    if (flowType === FlowType.AIGenerated) {
+                      const metaDescriptionEl = document.querySelector(
```

Comment:
Changed to to use refs instead. I initially was confused on how to properly do this since the ref needed to be passed into several components but found a clean solution to doing so.